File: EpiFib/mobile/EpiFib/EpiFib/EmergencyOwnerPacket.swift
//
//  EmergencyOwnerPacket.swift
//  EpiFib
//
//  Created by Gregory Hewett on 3/19/17.
//  Copyright © 2017 The Reach Lab, LLC. All rights reserved.
//

import UIKit
import EVReflection

class EmergencyOwnerPacket: EVNetworkingObject
{
    var emergencyInstanceId: String?
    
    var emergencyType: String?
        
    var ownerLocation: ExactLocation?
        
    var updatedTime: NSDate?
        
    var createdTime: NSDate?
        
    var ownerId: String?
        
    var nearbyContainers: [EmergencyContainer]?
        
    var numUsersNotified: NSNumber?
    
    var responderInfo: EmergencyUserInfo?

    convenience init(_
        emergencyInstanceId: String,
        emergencyType: String,
        ownerLocation: ExactLocation,
        updatedTime: NSDate,
        createdTime: NSDate,
        ownerId: String,
        nearbyContainers: [EmergencyContainer],
        numUsersNotified: NSNumber,
        responderInfo: EmergencyUserInfo)
    {
        self.init()
        self.emergencyInstanceId = emergencyInstanceId
        self.emergencyType = emergencyType
        self.ownerLocation = ownerLocation
        self.updatedTime = updatedTime
        self.createdTime = createdTime
        self.ownerId = ownerId
        self.nearbyContainers = nearbyContainers
        self.numUsersNotified = numUsersNotified
        self.responderInfo = responderInfo
    }
}

File: EpiFib/www/src/Common/Models/EmergencyOwnerPacket.cs
// <copyright file="EmergencyOwnerPacket.cs" company="The Reach Lab, LLC">
//     The Reach Lab, LLC. All rights reserved.
// </copyright>
// <author>Gregory Hewett</author>

namespace Common.Models
{
    using System;
    using System.Collections.Generic;

    public class EmergencyOwnerPacket : IEmergencyOwnerPacket
    {
        public string EmergencyInstanceId { get; set; }

        public EmergencyType EmergencyType { get; set; }

        public ExactLocation OwnerLocation { get; set; }

        public DateTime UpdatedTime { get; set; }

        public DateTime CreatedTime { get; set; }

        public string OwnerId { get; set; }

        public IEnumerable<EmergencyContainer> NearbyContainers { get; set; }

        public int NumUsersNotified { get; set; }

        public EmergencyUserInfo ResponderInfo { get; set; }

        public static explicit operator EmergencyOwnerPacket(EmergencyInstance v)
        {
            return new EmergencyOwnerPacket
            {
                EmergencyInstanceId = v.EmergencyInstanceId,
                EmergencyType = v.EmergencyType,
                NearbyContainers = v.NearbyContainers,
                NumUsersNotified = v.NumUsersNotified,
                OwnerId = v.OwnerId,
                OwnerLocation = v.OwnerLocation,
                ResponderInfo = v.ResponderInfo
            };
        }
    }
}
